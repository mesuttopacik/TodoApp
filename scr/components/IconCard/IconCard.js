import { View, Text } from 'react-native'
import React from 'react'
import  styles from './IconCard.style'
import Icon from 'react-native-vector-icons/dist/FontAwesome';

const IconCard = ({name}) => {
const circle = <Icon testID = 'list-icon-name' name={name} size={styles.size} color={styles.color} />;
  return (
    <View style={styles.container}>
    {circle}
  </View>
  )
}

export default IconCard