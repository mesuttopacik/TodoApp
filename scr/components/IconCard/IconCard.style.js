import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    container:{
        flexDirection:'row',
    },
    size:25,
    color:"#eeee",
})