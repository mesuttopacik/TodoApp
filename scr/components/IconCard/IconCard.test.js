import React from "react";
import { render } from "@testing-library/react-native";
import IconCard from "./IconCard";


test('should mach with snapshot', () => { 
    const comp = render(<IconCard/>)
    expect(comp).toMatchSnapshot()
 })

test('should icon title match sended title', () => { 
    const testName = 'times-circle-o'
    const comp = render(<IconCard name={testName}/>)
    const icon = comp.getByTestId('list-icon-name').props.style[0]

    expect(icon.fontSize).toBe(25)
    expect(icon.color).toBe('#eeee')
})