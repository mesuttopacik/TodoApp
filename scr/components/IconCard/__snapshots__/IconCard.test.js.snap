// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should mach with snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#eeee",
          "fontSize": 25,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="list-icon-name"
  >
    
  </Text>
</View>
`;
